app/models/game.rb

```diff
@@ -14,6 +14,30 @@ def populate!
     populate_right_white_half!
   end
 
+  def white_check?
+    white_king = pieces.find_by(type: 'King', color: 'white')
+    pieces.each do |piece|
+      if piece.enemy?(white_king) &&
+         piece.valid_move?(white_king.x_position,
```

Comment:
This logic is in #57 - prevent any king to step into any cell that is in the line of attack of opposite color piece, including the King.